cpp/src/core/radgeometry.h

```diff
@@ -0,0 +1,42 @@
+/*-------------------------------------------------------------------------
+*
+* File name:      radgeometry.h
+*
+* Project:        RADIA
+*
+* Description:    Geometry Results
+*
+* Author(s):      Robert Nagler
+*
+* First release:  2019
+* 
+* Copyright (C):  2019 by European Synchrotron Radiation Facility, France
+*                 All Rights Reserved
+*
+-------------------------------------------------------------------------*/
+
+#ifndef __RADGEOMETRY_H
+#define __RADGEOMETRY_H
+
+struct radGeometryFlattened {
+    int VerticesCount = 0;
+    double *Vertices = 0;
+    int Count = 0;
+    int *Lengths = 0;
+    float *Colors = 0;
+    ~radGeometryFlattened() {
+        // allocation is in PrepareGeomPolygDataForViewing
+        delete[] Vertices;
+        delete[] Lengths;
+        delete[] Colors;
+    }
+};
+
```

Comment:
It looks like the memory these pointers point to is allocated inside Radia shared library, however it is deallocated outside of that library (in the Py wrapper)? If so, this is not safe and has to be modified. I'll try to take care of this.

Reply:
This is only a problem on Windows. This code won't be running on Windows afaik.

Reply:
It might. VTK is a cross-platform package, and someone sooner or later may want/need to use this 3D viewer on Windows to view Radia geometries. I made Radia and SRW cross-platform some >20 years back (maybe that's why they survived:)) , and I'd like to keep them cross-platform.

Reply:
If you want to solve the DLL memory problem, you'll need to export an explicit destroy for `radGeometry` that is called before the return in `radia_ObjGeometry` (with a `try-catch` to prevent leaks in the case of exceptions). [Exporting C++ classes from a DLL](https://eli.thegreenplace.net/2011/09/16/exporting-c-classes-from-a-dll) is a good article.